File: CBuilder/EncryptDecrypt/Unit1.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Unit1.h"
#include "getcert.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "DemoBaseFormUnit"
#pragma link "clEncryptor"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
  : TclDemoBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnEncryptClick(TObject *Sender)
{
  TGetCertDialog *dlg = new TGetCertDialog(NULL);
  __try {
    dlg->Caption = "Get encryption certificate";
    if(dlg->ShowModal() == mrOk) {
      clEncryptor1->EncryptStore = dlg->edtStore->Text;
      clEncryptor1->EncryptCertificate = dlg->edtEmail->Text;
    }
  }
  __finally {
      delete dlg;
  }

  clEncryptor1->Encrypt(edtSource->Text, edtDestination->Text);

  ShowMessage(Format("The file %s was encrypted and stored to %s", ARRAYOFCONST((edtSource->Text, edtDestination->Text))));
}
//---------------------------------------------------------------------------

void __fastcall TForm1::btnDecryptClick(TObject *Sender)
{
  TGetCertDialog * dlg = new TGetCertDialog(NULL);
  __try {
    dlg->Caption = "Get encryption certificate";
    if(dlg->ShowModal() == mrOk) {
      clEncryptor1->EncryptStore = dlg->edtStore->Text;
      clEncryptor1->EncryptCertificate = dlg->edtEmail->Text;
    }
  }
  __finally {
      delete dlg;
  }

  clEncryptor1->Decrypt(edtSource->Text, edtDestination->Text);

  ShowMessage(Format("The file %s was decrypted and stored to %s", ARRAYOFCONST((edtSource->Text, edtDestination->Text))));
}
//---------------------------------------------------------------------------

void __fastcall TForm1::btnSourceClick(TObject *Sender)
{
  if (OpenDialog1->Execute())
    edtSource->Text = OpenDialog1->FileName;
}
//---------------------------------------------------------------------------

void __fastcall TForm1::btnDestinationClick(TObject *Sender)
{
  if (SaveDialog1->Execute())
    edtDestination->Text = SaveDialog1->FileName;
}
//---------------------------------------------------------------------------


void __fastcall TForm1::clEncryptor1Progress(TObject *Sender,
      __int64 ABytesProceed, __int64 ATotalBytes)
{
  ProgressBar1->Max = ATotalBytes;
  ProgressBar1->Position = ABytesProceed;
}
//---------------------------------------------------------------------------

File: Delphi/EncryptDecrypt/Unit1.pas
unit Unit1;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, ComCtrls, clEncryptor, getcert, DemoBaseFormUnit, ExtCtrls;

type
  TForm1 = class(TclDemoBaseForm)
    btnEncrypt: TButton;
    btnDecrypt: TButton;
    Label1: TLabel;
    Label2: TLabel;
    Label3: TLabel;
    edtSource: TEdit;
    btnSource: TButton;
    edtDestination: TEdit;
    btnDestination: TButton;
    OpenDialog1: TOpenDialog;
    SaveDialog1: TSaveDialog;
    ProgressBar1: TProgressBar;
    clEncryptor1: TclEncryptor;
    procedure btnSourceClick(Sender: TObject);
    procedure btnDestinationClick(Sender: TObject);
    procedure btnEncryptClick(Sender: TObject);
    procedure btnDecryptClick(Sender: TObject);
    procedure clEncryptor1Progress(Sender: TObject; ABytesProceed,
      ATotalBytes: Int64);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.dfm}

procedure TForm1.btnSourceClick(Sender: TObject);
begin
  if OpenDialog1.Execute() then
  begin
    edtSource.Text := OpenDialog1.FileName;
  end;
end;

procedure TForm1.btnDestinationClick(Sender: TObject);
begin
  if SaveDialog1.Execute() then
  begin
    edtDestination.Text := SaveDialog1.FileName;
  end;
end;

procedure TForm1.btnEncryptClick(Sender: TObject);
var
  dlg: TGetCertDialog;
begin
  dlg := TGetCertDialog.Create(nil);
  try
    dlg.Caption := 'Get encryption certificate';
    if (dlg.ShowModal() = mrOK) then
    begin
      clEncryptor1.EncryptStore := dlg.edtStore.Text;
      clEncryptor1.EncryptCertificate := dlg.edtEmail.Text;
    end;
  finally
    dlg.Free();
  end;

  clEncryptor1.Encrypt(edtSource.Text, edtDestination.Text);

  ShowMessage(Format('The file %s was encrypted and stored to %s', [edtSource.Text, edtDestination.Text]));
end;

procedure TForm1.btnDecryptClick(Sender: TObject);
var
  dlg: TGetCertDialog;
begin
  dlg := TGetCertDialog.Create(nil);
  try
    dlg.Caption := 'Get encryption certificate';
    if (dlg.ShowModal() = mrOK) then
    begin
      clEncryptor1.EncryptStore := dlg.edtStore.Text;
      clEncryptor1.EncryptCertificate := dlg.edtEmail.Text;
    end;
  finally
    dlg.Free();
  end;

  clEncryptor1.Decrypt(edtSource.Text, edtDestination.Text);

  ShowMessage(Format('The file %s was decrypted and stored to %s', [edtSource.Text, edtDestination.Text]));
end;

procedure TForm1.clEncryptor1Progress(Sender: TObject; ABytesProceed, ATotalBytes: Int64);
begin
  ProgressBar1.Max := ATotalBytes;
  ProgressBar1.Position := ABytesProceed;
end;

end.
